THREESCALE-1789: Add "Managed by operator" banner (UI)

jlledom opened the pull request:
**What this PR does / why we need it**:

This replaces https://github.com/3scale/porta/pull/3865. That got closed after merging https://github.com/3scale/porta/pull/3857

The operator can manage porta resources through API. When a resource is managed by operator, we must inform the user about that.

https://github.com/3scale/porta/pull/3857 implements the backend support (annotations). This PR implements the frontend, for now, just a banner in the relevant screens:

- `http://provider-admin.3scale.localhost:3000/buyers/accounts/4`
- `http://provider-admin.3scale.localhost:3000/buyers/accounts/4/edit`
- `http://provider-admin.3scale.localhost:3000/apiconfig/services/2`
- `http://provider-admin.3scale.localhost:3000/apiconfig/services/2/edit`
- `http://provider-admin.3scale.localhost:3000/p/admin/backend_apis/2`
- `http://provider-admin.3scale.localhost:3000/p/admin/backend_apis/2/edit`

Examples:

![image](https://github.com/user-attachments/assets/3c053c24-ae8a-474e-8984-6d0ceca46945)


![image](https://github.com/user-attachments/assets/31f0863e-1bad-4412-b01e-57dfd485dc87)


**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-1786

**Verification steps** 

1. Marks a resource as managed by operator. Check https://github.com/3scale/porta/pull/3857 to know how
2. Visit the resource `#show` or `#edit` screens
3. You should see the banner

josemigallas commented:
It looks good like that, but I would rather follow Patternfly's design guidelines regarding inline alerts: https://www.patternfly.org/components/alert/design-guidelines#plain-inline-alerts:

![](https://www.patternfly.org/images/inline-header-alert.0b3fb4ed0a9c01b9.png)

jlledom commented on app/views/api/services/edit.html.slim:
```diff
@@ -1,3 +1,6 @@
+-content_for :annotation do
```
Ouch, sorry...

jlledom commented:
> It looks good like that, but I would rather follow Patternfly's design guidelines regarding inline alerts: https://www.patternfly.org/components/alert/design-guidelines#plain-inline-alerts:

I made some changes. Now it looks like this:

![image](https://github.com/user-attachments/assets/20292417-7c5c-42d2-818d-3d65bbcf9b73)

Now it uses a plan inline alert but there's no spacing between the title and the alert. I've been trying to fix it but I'm spending too much time and I think its better for somebody else with more experience to do it... @josemigallas

jlledom commented:
@josemigallas  I made some changes, I'd say it looks better now:

![image](https://github.com/user-attachments/assets/83f52353-587c-48d6-9dd6-728ef16a3049)

![image](https://github.com/user-attachments/assets/94088790-ca0f-40a3-b13b-b71094f9eee6)

akostadinov commented:
Maybe make orange or something? Looks good. Just I'm not sure anybody would notice blue stuff :) Blue means all is fine. idk. Just saying.

jlledom commented:
> Maybe make orange or something? Looks good. Just I'm not sure anybody would notice blue stuff :) Blue means all is fine. idk. Just saying.

It would look like this:

![image](https://github.com/user-attachments/assets/0ffdc5bb-6c8c-45cf-8827-f63a42d40e6b)

WDYT @josemigallas 

We can also have it as `info` for the `#show` action and `warning` for the `#edit` action

akostadinov commented:
Better to me. But just an opinion, do as you want.

josemigallas commented:
With the description it now looks weird 😅. Maybe we don't use `.pf-m-plain` when there is a description? Or never.

> We can also have it as info for the #show action and warning for the #edit action

On the other hand, I agree with this ☝🏻

jlledom commented:
After the last changes @josemigallas @akostadinov 

![image](https://github.com/user-attachments/assets/fad19253-06f0-45e7-b7d8-a14606734670)

![image](https://github.com/user-attachments/assets/acea5949-8738-48e1-9546-2db104f15652)

lvillen commented:
I think we could use the `inline alert` also at the #show action, which would make code climate happy, but LGTM.

jlledom commented:
> I think we could use the `inline alert` also at the #show action, which would make code climate happy, but beside that LGTM.

I added the `plain` style to the `#show` action as suggested by @josemigallas so I think we should keep it. However, I refactored the code to (hopefully) remove the warning from codeclimate: https://github.com/3scale/porta/pull/3876/commits/c5c1999c5ac0963f825ff61c159d83e87407e8b9

lvillen reviewed (approved):
🚀
